packages/api/game/types.go

```diff
@@ -0,0 +1,140 @@
+package game
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/google/uuid"
+	"github.com/gorilla/websocket"
+)
+
+type BaseMessage struct {
+	Event string          `json:"event"`
+	Data  json.RawMessage `json:"data"`
+}
+
+type PlayerJoinData struct {
+	name string
+	x    float32
+	y    float32
+}
+
+type PlayerMoveData struct {
+	id string
+	x  float32
+	y  float32
+}
+
+type IdData struct {
+	id string
+}
+
+type ItemEntityCreateData struct {
+	item string
+	x    float32
+	y    float32
+}
+
+type Game struct {
+	ctx          context.Context
+	connection   *websocket.Conn
+	players      []*Player
+	itemEntities []*ItemEntity
+}
+
+func (g *Game) HandlePlayerJoin(data PlayerJoinData) {
+	fmt.Println("player join")
+
+	player := &Player{
+		id:   uuid.New(),
+		name: data.name,
+		x:    data.x,
+		y:    data.y,
+	}
+
+	playersPatched := append(g.players, player)
+	fmt.Println("players patched: ", playersPatched)
+}
+
+func (g *Game) HandlePlayerLeave(data IdData) {
+	fmt.Println("player leave")
+
+	for i, player := range g.players {
+		uuid, err := uuid.FromBytes([]byte(data.id))
+		if err != nil {
+			fmt.Println("could not parse uuid: ", err)
```

Comment:
should this method return an error?

Reply:
small mistake on my side. everyone is pretty wrong in that method actually because I shouldn't even be parsing the data uuid in the for loop, I can just parse it outside of it and return if there is an error